src/pykeen/training/training_loop.py

```diff
@@ -585,14 +596,16 @@ def _train(  # noqa: C901
                         stop = min(start + sub_batch_size, current_batch_size)
 
                         # forward pass call
-                        current_epoch_loss += self._forward_pass(
+                        batch_loss = self._forward_pass(
                             batch,
                             start,
                             stop,
                             current_batch_size,
                             label_smoothing,
                             slice_size,
                         )
+                        current_epoch_loss += batch_loss
+                        callback.on_batch(epoch=epoch, batch_loss=batch_loss)
```

Comment:
should we also pass the batch here?

Reply:
Maybe we can add `**kwargs` to all callbacks so we can later on add other things without breaking old callbacks?

Then, I would wait until we actually need it in at least one callback,

Reply:
Good idea. I had considered that